bot/delete_log.py

```diff
@@ -0,0 +1,136 @@
+import discord
+from discord.ext.commands import command, Cog, Context
+from typing import TYPE_CHECKING, Dict, Optional
+import re
+from bot.vote_session import VoteSessionBase
+
+if TYPE_CHECKING:
+    from bot.main import RedstoneSquid
+
+APPROVE_EMOJI = "✅"
+DENY_EMOJI = "❌"
+
+
+class DeleteLogSession(VoteSessionBase):
+    def __init__(
+        self,
+        message: discord.Message,
+        target_message: Optional[discord.Message] = None,
+        threshold: int = 7,
+    ):
+        super().__init__(message, threshold)
+        self.target_message = target_message
+
+
+class DeleteLogCog(Cog, name="Vote"):
+    def __init__(self, bot: "RedstoneSquid"):
+        self.bot = bot
+        self.tracked_messages: Dict[int, DeleteLogSession] = {}
+
+    @command(name="start_vote")
+    async def start_vote(self, ctx: Context, message_url: Optional[str] = None):
```

Comment:
just use a [converter](https://discordpy.readthedocs.io/en/stable/ext/commands/commands.html#discord-converters) for message_url